classes/GoogleAnalyticsTools.php

```diff
@@ -42,38 +42,39 @@ public function filter($gaScripts, $filterable)
     }
 
     /**
-     * add order transaction
+     * Renders purchase event for order
      *
-     * @param array $products
-     * @param array $transaction
+     * @param array $orderProducts
+     * @param array $orderData
+     * @param string $callbackUrl
      *
      * @return string|void
      */
-    public function addTransaction($products, $transaction)
+    public function renderPurchaseEvent($orderProducts, $orderData, $callbackUrl)
```

Comment:
```suggestion
    public function renderPurchaseEvent(array $orderProducts, array $orderData, string $callbackUrl)
```
New function? Let's make it as clean as possible from day one 😉

Reply:
We cant use named arguments because Google Analytics supports Prestashop versions with low PHP support. I think it's not worth to break the compatibility if there is no other benefit. 👍